clef/clef.go

```diff
@@ -6,15 +6,37 @@ import (
 	"github.com/hosslen/lfd/eardet"
 	"github.com/hosslen/lfd/murmur3"
 	"github.com/hosslen/lfd/rlfd"
+
+	"github.com/hosslen/lfd/cuckoo"
 )
 
+type leakyBucket struct {
+	//moment in time when flow was inserted into watchlist
+	firstTimestamp time.Duration
+	//moment in time when last packet for this flow was received
+	lastTimestamp time.Duration
+	//how many bytes the bucket contains right now
+	count uint32
+}
+
 type pktTriple struct {
 	flowID uint32
 	size uint32
 	t time.Duration
 }
 
 type ClefDtctr struct {
+
+	//flow lists
+	watchlist map[uint32](*leakyBucket)
+	maxWatchlistSize uint32
+	blacklist *cuckoo.CuckooTable
+
+	//flow specification
+	beta float64
+	gamma float64
+	t_l time.Duration
```

Comment:
Actually `t_l` is not a part of flow spec. The flow spec only contains beta and gamma which are all parameters for a leaky bucket model. The RLFD actually cannot guarantee exactly no false negative over the flow spec, because RLFD has the `t_l` and only check the counter every `t_l` time intervals. I think we should say t_l is a part of setting in RLFD rather than flow spec.

So if this `t_l` for watchlist is the same as the `t_l` in `rlfd1`, we can just fetch t_l from the `rlfd1` instance, so that we can avoid some accident inconsistency, when we are using the CLEF.

Note that, if we say the `t_l` in `rlfd1` is `t_l_1` and the `t_l` in `rlfd2` is `t_l_2`, the `t_l_2 > t_l_1` by our design. If you have our CLEF paper draft for CCS'17, you can see Section 4.5 paragraph of `Twin-RLFD parallel composition` for motivation.